deploy-agent/deployd/common/helper.py

```diff
@@ -65,17 +65,26 @@ def get_stale_builds(build_timestamps, num_builds_to_retain=2):
     def clean_package(base_dir, builds, build_name):
         local_fn = '{}-{}.*'.format(build_name, builds)
         try:
+            # Remove extracted pointer from disk
             extracted_file = os.path.join(base_dir, '{}.extracted'.format(builds))
             if os.path.exists(extracted_file):
                 os.remove(extracted_file)
+            # Remove staged pointer from disk
             staged_file = os.path.join(base_dir, '{}.staged'.format(builds))
             if os.path.exists(staged_file):
                 os.remove(staged_file)
         except OSError as e:
             log.error(e)
 
         try:
+            # Remove build directory from disk
             shutil.rmtree(os.path.join(base_dir, builds))
+        except BaseException as e::
+            # Catch base exception class, as there's a mulitude of reasons a rmtree can fail
+            log.error(e)
```

Comment:
How about using log.exception, see:
https://stackoverflow.com/questions/5191830/best-way-to-log-a-python-exception

Reply:
👍 Will update PR

Reply:
typo in comment line 83: multitude

Reply:
2 hours of sleep will do that - fixing